test/integration/test_use_proxy.py

```diff
@@ -0,0 +1,80 @@
+"""Integration test for agent-daemon communication through MITM proxy."""
+import asyncio
+from functools import partial
+from pathlib import Path
+
+import pytest
+
+
+from goth.configuration import load_yaml
+from goth.runner import Runner
+from goth.runner.probe import RequestorProbe
+
+from goth.assertions import EventStream
+from goth.api_monitor.api_events import APIEvent, APIRequest
+
+
+async def api_call_made(container_name: str, stream: EventStream[APIEvent]) -> bool:
+    """Assert that an API call to `container_name` has been made in the past."""
+
+    for event in stream.past_events:
+        if isinstance(event, APIRequest) and event.callee == f"{container_name}:daemon":
+            return True
+    raise AssertionError(f"No API call to {container_name} registered by proxy")
+
+
+async def no_api_call_made(container_name: str, stream: EventStream[APIEvent]) -> bool:
+    """Assert that no API call to `container_name` has been made in the past."""
+
+    try:
+        await api_call_made(container_name, stream)
+    except AssertionError:
+        return True
+    raise AssertionError(f"API call to {container_name} registered by proxy")
+
+
+@pytest.mark.asyncio
+async def test_use_proxy(default_goth_config: Path, log_dir: Path) -> None:
+    """Test if runner correctly sets up agent-deamon communication through proxy."""
+
+    overrides = [
+        (
+            "nodes",
+            [
+                {"name": "requestor", "type": "Requestor", "use-proxy": True},
+                {"name": "provider-1", "type": "VM-Wasm-Provider", "use-proxy": True},
+                {"name": "provider-2", "type": "VM-Wasm-Provider", "use-proxy": False},
+            ],
+        )
+    ]
```

Comment:
I'm just trying to set `use-proxy` on `provider-2` to `False`, perhaps there's an easier way to do it?

Reply:
OTOH making the "nodes" part of the config so explicit has its advantages -- it's explicit.